dotnet/samples/KernelSyntaxExamples/Example70_Agents.cs

```diff
@@ -54,22 +55,29 @@ public Task RunSimpleChatAsync()
     /// Tools/functions: MenuPlugin
     /// </summary>
     [Fact]
-    public Task RunWithMethodFunctionsAsync()
+    public async Task RunWithMethodFunctionsAsync()
     {
         WriteLine("======== Run:WithMethodFunctions ========");
 
-        KernelPlugin plugin = KernelPluginFactory.CreateFromType<MenuPlugin>();
+        MenuPlugin menuApi = new();
+        KernelPlugin plugin = KernelPluginFactory.CreateFromObject(menuApi);
 
         // Call the common chat-loop
-        return ChatAsync(
+        await ChatAsync(
             "Agents.ToolAgent.yaml", // Defined under ./Resources/Agents
             plugin,
-            arguments: null,
+            arguments: new() { { MenuPlugin.CorrelationIdArgument, Guid.NewGuid() } },
```

Comment:
Suggestion: Instead of passing a GUID generated at runtime, we could pass a 'known' GUID (literal) and verify that we are getting the same GUID back.

Reply:
Updated with: 3.141592653